google/cloud/storage/tests/object_integration_test.cc

```diff
@@ -50,24 +50,68 @@ class ObjectIntegrationTest : public ::testing::Test {
            ".txt";
   }
 
+  std::string LoremIpsum() const {
+    return R"""(Lorem ipsum dolor sit amet, consectetur adipiscing
+elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim
+ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
+commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit
+esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
+non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+})""";
+  }
+
  protected:
   google::cloud::internal::DefaultPRNG generator_ =
       google::cloud::internal::MakeDefaultPRNG();
 };
 }  // anonymous namespace
 
+/// @test Verify the Object CRUD (Create, Get, Update, Delete, List) operations.
+TEST_F(ObjectIntegrationTest, BasicCRUD) {
+  gcs::Client client;
+  auto bucket_name = ObjectTestEnvironment::bucket_name();
+
+  auto objects = client.ListObjects(bucket_name);
+  std::vector<gcs::ObjectMetadata> initial_list(objects.begin(), objects.end());
+
+  auto name_counter = [](std::string const& name,
+                         std::vector<gcs::ObjectMetadata> const& list) {
+    auto name_matcher = [](std::string const& name) {
+      return [name](gcs::ObjectMetadata const& m) { return m.name() == name; };
+    };
+    return std::count_if(list.begin(), list.end(), name_matcher(name));
+  };
+
+  auto object_name = MakeRandomObjectName();
+  ASSERT_EQ(0, name_counter(object_name, initial_list))
+      << "Test aborted. The object <" << object_name << "> already exists."
+      << "This is unexpected as the test generates a random object name.";
+
+  // Create the object, but only if it does not exist already.
+  auto insert_meta = client.InsertObject(bucket_name, object_name, LoremIpsum(),
```

Comment:
I think it's worth typing out the return type in a lot of places (in tests especially) instead of using auto.  W.r.t. this line: some client types return a {Status, <type>} pair, while others just return a <type> object -- I had to verify which client the test was using and which behavior it went with while I was typing this :)